### ISSUE TITLE
[Feature] Extend from a base config

### ISSUE THREAD

**Issue by ematipico** [webpack/webpack-cli#569]
**Is your feature request related to a problem? Please describe.**
Not a problem, but a nice to have.
Lot of time people have different configuration file based on different environment or situation and sometimes they like to put the 'shared' configuration inside a specific file (`webpack.base.js`, something like this). It would nice if the `webpack-cli` would provide an automatic way to extend from a base file.

**Describe the solution you'd like**

Usage scenarios:

```js
// webpack.base.js
module.exports = {
  entry: {
     foo: './src/foo.js',
     bar: './src.bar.js'
   },
   output: {
      filename: [name].js
   }
}
```

```js
// webpack.dev.js

module.exports = {
   mode: 'development',
   plugins: [
       new DefinePlugin() // anything
   ]
}
```

```bash
webpack --config webpack.dev.js --extend webpack.base.js
```

Or 

```js
// inside webpack.dev.js

module.exports = {
   extend: './webpack.base.js',
   mode: 'development',
   plugins: [
       new DefinePlugin() // anything
   ]
}
```

**Comment by EugeneHlushko** [comment]
Do you mean like this? https://github.com/survivejs/webpack-merge

**Comment by sumit-gupta91** [comment]
I would like to work on this feature, if no one has taken this up already.

**Comment by ev1stensberg** [comment]
Feel free to 🤘🏽

**Comment by xavierfoucrier** [comment]
As seen in webpack-command: https://github.com/webpack-contrib/webpack-command#extendable-webpack-configurations :+1:

**Comment by ev1stensberg** [comment]
This is possible. Gonna do this in the new cli version

**Comment by wilkinsonjack1993** [comment]
I really like the way this is done by the now deprecated config-loader: https://webpack.js.org/loaders/config-loader/#extending-configuration-files 

Its really clean and obvious which files are extending what. If webpack-cli could do this in a similar way that would be awesome.

**Comment by ev1stensberg** [comment]
Send a PR and it could be added pretty fast 🤙🏽

**Comment by artola** [comment]
I wrongly reported this issue into `webpack` repo ([issue](https://github.com/webpack/webpack/issues/8119#issuecomment-426213075)).

Most of the tools in the stack support "extends", even it was available on "webpack-command", and today shown in webpack documentation as possible (but it is not).

Examples: `tslint, `jest`, `eslint`, `babel`.

**Comment by ematipico** [comment]
It is possible via `config-loader`. What's need to be done in here and to use via cli and also being able to use `config-loader` inside our code. Like webpack command does

**Comment by evenstensberg** [comment]
What this consists of is to merge the configuration you are provided as an option with this one given as a flag

**Comment by evenstensberg** [comment]
This is being resolved at #527

**Comment by xavierfoucrier** [comment]
I am now using [`webpack-merge`](https://www.npmjs.com/package/webpack-merge) as an alternative but it would be cool to have this feature integrated the same way `webpack-command` CLI did with the [`extends`](https://github.com/webpack-contrib/webpack-command#extendable-webpack-configurations) property.
